### ISSUE TITLE
Investigate alternative notion of managed resources than `app.kubernetes.io/instance` label 

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-cd#1482]

**Comment by jessesuen** [comment]
For more context. See:
* https://github.com/argoproj/argo-cd/issues/1066#issuecomment-468893285
* https://github.com/argoproj/argo-cd/issues/1066#issuecomment-499310959

Essentially this issue is for us to find an alternative way to detect what is considered a "managed" resource of an application, which wouldn't involve label injection.

The issue that we currently face with the `app.kubernetes.io/instance` label is that there is friction about using this label with other common tooling such as: 
* helm - which started using `app.kubernetes.io/instance` as part of their default template generation, including using `app.kubernetes.io/instance` in selectors.
* kustomize - which has a `commonLabels` feature that **_also_** potentially injects `app.kubernetes.io/instance` into selectors (provided the user specifies `app.kubernetes.io/instance` in their commonLabels).

Since v0.11, Argo CD no longer requires that the resource association to be driven by a label. Previously we depended on a label so that we could perform efficient kubernetes queries to perform resource discovery. Now, however, our live state metadata cache enables us to efficiently discover all resources associated with an application, without the use of a label. We simply need to inject the application name **_somewhere_** in the resource object. This could be a:
* label (what we have today)
* annotation
* ownerReference (assuming cross-cluster ownerReference is valid)

In summary, our injection of `app.kubernetes.io/instance` seems that it may be too opinionated, causing the problems.

My ideal solution would be to inject an ownerReference to all managed resources that Argo CD deploys. However, I do not know if this is valid thing to do when the "owner" is a resource in an entirely different cluster. I posed this question to sig-api-machinery, and still waiting a response.

**Comment by alexec** [comment]
Note we might wish to use `argocd.argoproj.io/instance` for this as that is our convention.

**Comment by alexec** [comment]
@jessesuen did you get a response to your query in sig-api-machinery

**Comment by alexec** [comment]
Document work-aroindp

**Comment by yann-soubeyrand** [comment]
Hi,
Shouldn't Argo CD use the common `app.kubernetes.io/managed-by` label (https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/)?

**Comment by johnmarcou** [comment]
@yann-soubeyrand The `app.kubernetes.io/managed-by` label does make sense.
The one used by helm is only print at the templating, but never use this helm never manage the application.
